provider/pkg/schema/gen.go

```diff
@@ -1225,6 +1225,12 @@ func (ctx *cfSchemaContext) genEnumType(enumName string, propSchema *jsschema.Sc
 	case "networkfirewall:RuleGroupType":
 		typName = "RuleGroupTypeEnum" // Go SDK name conflict vs. RuleGroup resource
 	}
+
+	switch ctx.cfTypeName + "   " + enumName {
```

Comment:
Why is this specifically 3 whitespaces?

Reply:
No reason, just need some sort of separator since I can't pattern match a  tuple (a, b) against a pattern (a, b) in Go.

Reply:
This an odd thing to test since it's not quite a feature, it's a data correction. The schemagen decides how to map cfTypeName to Pulumi type token with some degrees of freedom, and we're adding some configuration here to direct it better. I think it will be obvious if this works in subsequent PR that should avoid the panic. If this happens often we should make this a true feature of the generator where it can source config data that informs this name mapping, and test that.